README.md

```diff
@@ -1,5 +1,125 @@
 # Kafka Histo Streamer
 
-Consumes event data from kafka and re-streams it as histogram data
+This is a prototype to test and demonstrate Mantid being capable of live processing event data into histogram data, streaming both via Kafka. Subscribes to a topic with event data and publishes it to another topic as histogram data.
 
+## Requirements
 
+Requires Mantid (currently has to be `nightly`).
+
+Virtualenv Python package recommended and used below (`$ pip install virtualenv`).
+
+Running a local demo additionally requires [Docker](https://www.docker.com/) and [ISIS Nexus Streamer](https://github.com/ScreamingUdder/isis_nexus_streamer_for_mantid/).
+
+Already includes the `hs00` schema, and generated code, from the DMSC [Schema Repository](https://github.com/ess-dmsc/streaming-data-types).
+
+
+## Setup
+
+Ensure [Mantid](https://www.mantidproject.org/Main_Page) is installed.
+
+Clone this repository:
+
+```
+$ git clone https://github.com/DMSC-Instrument-Data/kafka_histo_streamer
+```
+
+
+You will want to edit `run.py` and `postprocess.py` to configure `kafka_server` and `kafka_topic` as needed. If you want to run a local demo, we will revisit what these should be set to below.
```

Comment:
"as needed" is a bit confusing. Maybe elaborate on (or just link to) what can be changed.

Reply:
I thought about elaborating here as well, but the only things that can be changed are `kafka_server` and `kafka_topic`, which seem fairly self-explanatory... not sure what I could add here?

Alternatively, I thought about pulling these out to a `config.py` or `settings.py`. That way all user configurable parameters are in one place, and I could add some brief descriptions there as comments. What do you think?

Reply:
The way I see it, this is about streaming neutron data, and kafka is an implementation detail. Therefore being as kind as possible to future readers might just involve changing this to say: 

You will want to edit `run.py` and `postprocess.py` to configure `kafka_server` (IP:Port) and `kafka_topic` 

**(by the way, `kafka_topic` is not something that the run script actually has a variable for)** Linked to InstrumentName?

Reply:
The `run.py` script launches the consumption of event data, whereas `postprocess.py` is responsible for pushing the histogram data.

The Kafka topic where the event data is pulled from is always named after the instrument plus a suffix. The suffixes are defined as constants in [KafkaTopicSubscriber.cpp](https://github.com/mantidproject/mantid/blob/master/Framework/LiveData/src/Kafka/KafkaTopicSubscriber.cpp#L69-L73). Effectively, this topic name is fixed for any given instrument and can never change. From my talks with Matt, I understand this wasn't always the case (at least there used to be an override to manually specify a topic name) but was intentionally changed to be this way, by design (and causes some frustration for debugging and development).

Similarly, the output topic where we publish the histograms (the one in `postprocess.py`) *must* be named after the instrument plus the suffix defined for histogram data in KafkaTopicSubscriber.cpp. Otherwise, Mantid's KafkaHistoListener will not be able to find it (since, again, it is fixed and not overridable). If these naming conventions change, then any instructions I write about these details here would be invalidated.

With the above in mind, I think it would be more correct to say that this is primarily about streaming data over Kafka within a very specific context that imposes constraints outside of our control. From this perspective, I'd say the fact that the data is neutron data is the implementation detail.

Whoever sets this up, runs it, or does further work on it needs to be intimately familiar with details of the conventions and layout of the Kafka infrastructure at the ESS, much more so than knowing anything about neutron data. I wrote the readme with this audience in mind.